web/src/rendered_markdown.ts

```diff
@@ -345,7 +345,13 @@ export const update_elements = ($content: JQuery): void => {
             const $inline_img_thumbnail = $(this);
             let thumbnail_name = thumbnail.preferred_format.name;
             if ($inline_img_thumbnail.attr("data-animated") === "true") {
-                thumbnail_name = thumbnail.animated_format.name;
+                if (user_settings.web_animate_image_previews === "always") {
+                    thumbnail_name = thumbnail.animated_format.name;
+                } else {
+                    $inline_img_thumbnail
+                        .closest(".message_inline_image")
+                        .addClass("message_inline_animated_image_still");
```

Comment:
Added the requested behavior of treating "hover" as "always" on mobile devices.

``` diff
diff --git a/web/src/rendered_markdown.ts b/web/src/rendered_markdown.ts
index aa19e7cf69..9081b7e7c9 100644
--- a/web/src/rendered_markdown.ts
+++ b/web/src/rendered_markdown.ts
@@ -345,9 +345,18 @@ export const update_elements = ($content: JQuery): void => {
             const $inline_img_thumbnail = $(this);
             let thumbnail_name = thumbnail.preferred_format.name;
             if ($inline_img_thumbnail.attr("data-animated") === "true") {
-                if (user_settings.web_animate_image_previews === "always") {
+                if (user_settings.web_animate_image_previews === "always" ||
+                    (user_settings.web_animate_image_previews === "on_hover" &&
+                        util.is_mobile())) {
+                    // Use a constantly animating thumbnail in the
+                    // "always" configuration, or the
+                    // "on_hover" configuration on mobile devices,
+                    // where hovering is impossible and there's much
+                    // less on the screen.
                     thumbnail_name = thumbnail.animated_format.name;
                 } else {
+                    // If we're showing a still thumbnail, show a play
+                    // button so that users that it can be played.
                     $inline_img_thumbnail
                         .closest(".message_inline_image")
                         .addClass("message_inline_animated_image_still");
```